docs/DeveloperGuide.md

```diff
@@ -85,6 +85,74 @@ This section describes some noteworthy details on how certain features are imple
 
+TrackIt@NUS allows users to keep track of his/her tasks. The task manager is one of TrackIt@NUS's basic operations
+.
+ 
+ #### Rationale
+ 
+ The task manager is a crucial feature in TrackIt@NUS. Tasks are an integral part of any student's day-to-day life
+ . Hence, TrackIt@NUS includes a task manager for students to keep track of all their tasks. To better
+  support NUS students, a task can either belong to a module or not. When adding a task, users
+  can choose to the include the `m/MODULE_CODE` parameter in order to add a task that belongs to a module. When users
+   click into a specific module tab, they can see the tasks belonging to each module.
+   
+![ModuleTasks](images/ModuleTasks.png)
+    
+:information_source: A task does not have to belong a module. In this case, the module parameter of the task is
+ simply treated as null.
+     
+:warning: The module must exist (i.e. there must be a module with the specified `MODULE_CODE`) otherwise the add
+ command will not work.
+ 
+:bulb: To remove a task from a module, simply type `T edit INDEX m/` (use the `m/` prefix but leave the `MODULE_CODE
+` parameter empty).
+
+#### Current Implementation
+
+In this section, we will outline the key commands of the Task Manager, namely:
+* `AddTaskCommand`
+* `DeleteTaskCommand`
+* `EditTaskCommand`
+
+We will also elaborate on one more key function, namely `getModuleTasks`.
+
+Generally, this is how the Task Manager handles a command.
+ 
+ ![Add Task](images/AddTaskActivityDiagram.png)
+ 
+The add, delete, and edit commands are all implemented in similar ways. When they are executed they will:
+ * call on the relevant Model methods
+ * update the `UniqueTaskList` depending on the command
+ * Save the updated task list to `data/trackIter.json`
+ * return the relevant CommandResult message
+ 
+ The following shows the sequence diagram of the `AddTaskCommand`.
+
+![Add Task Sequence Diagram](images/AddTaskSequenceDiagram.png)
+
+The following shows the sequence diagram of the `DeleteTaskCommand`.
+
+![Delete Task Sequence Diagram](images/DeleteTaskSequenceDiagram.png)
+
+The follow shows the sequence diagram of the `EditTaskCommand`.
+
+![Edit Task Sequence Diagram](images/EditTaskSequenceDiagram.png)
+
+The `getModuleTasks` function takes in a Module Code and returns all tasks that belong to the specified module code
+. It achieves this by:
+* Retrieving all tasks in TrackIt@NUS 
+* Filtering the out the tasks based on module code
+
+This is the sequence diagram of `getModuleTasks`.
+
+![Get Module Tasks Sequence Diagram](images/GetModuleTasksSequenceDiagram.png)
+
+#### Design Considerations
+
+Just by observing the above 3 sequence diagrams, it is clear that adding, deleting, and editing a task are
+ implemented in very similar ways. We chose to do this for simplicity. All task-related commands should behave the
+  same way and hence it is easier for developers to learn the code base and make contributions. 
+
```

Comment:
I think a better rationale(or at least better sounding haha) is that we implement all commands in a very similar way so that the logic is uniformed and also for better maintainability / less prone to bugs. 

But I don't know should we place this here or put it in a more "general" section of this DG? And put more task-specific design consideration here?

Reply:
I think we can put this in th e`Architecture` section of the Dg